timed/employment/views.py

```diff
@@ -53,30 +53,31 @@ class UserViewSet(ModelViewSet):
 
     def get_queryset(self):
         user = self.request.user
-        current_employment = models.Employment.objects.get_at(
-            user=user, date=datetime.date.today()
-        )
         queryset = get_user_model().objects.prefetch_related(
             "employments", "supervisees", "supervisors"
         )
 
-        if current_employment.is_external:
-            assigned_tasks = Task.objects.filter(
-                Q(task_assignees__user=user, task_assignees__is_reviewer=True)
-                | Q(
-                    project__project_assignees__user=user,
-                    project__project_assignees__is_reviewer=True,
+        if user.employments.all().exists():
```

Comment:
This checks whether a user *had* an employment *at any time*, which is incorrect. You need to call the `get_at()` method as you do within the `if` statement anyway to figure out if the user has an employment. You can use `try/except` to handle the case where the user doesn't have an active employment.

Reply:
Makes sense. I refactored the code to use `try/except`